compute/cloud-client/src/main/java/compute/preemptible/CreatePreemptibleInstance.java

```
@@ -0,0 +1,113 @@
+    long diskSizeGb = 10L;
+    String networkName = "default";
+
+    try (InstancesClient instancesClient = InstancesClient.create()) {
+
+      AttachedDisk disk =
+          AttachedDisk.newBuilder()
+              .setBoot(true)
+              .setAutoDelete(true)
+              .setType(AttachedDisk.Type.PERSISTENT.toString())
+              .setInitializeParams(
+                  // Describe the size and source image of the boot disk to attach to the instance.
+                  AttachedDiskInitializeParams.newBuilder()
+                      .setSourceImage(sourceImage)
+                      .setDiskSizeGb(diskSizeGb)
+                      .build())
+              .build();
+
+      // Use the default VPC network.
+      NetworkInterface networkInterface = NetworkInterface.newBuilder()
+          .setName(networkName)
+          .build();
+
+      // Collect information into the Instance object.
+      Instance instanceResource =
+          Instance.newBuilder()
+              .setName(instanceName)
+              .setMachineType(machineType)
+              .addDisks(disk)
+              .addNetworkInterfaces(networkInterface)
+              // Set the preemptible setting.
+              .setScheduling(Scheduling.newBuilder()
+                  .setPreemptible(true)
+                  .build())
+              .build();
+
+      System.out.printf("Creating instance: %s at %s %n", instanceName, zone);
+
+      // Prepare the request to insert an instance.
+      InsertInstanceRequest insertInstanceRequest = InsertInstanceRequest.newBuilder()
+          .setProject(projectId)
+          .setZone(zone)
+          .setInstanceResource(instanceResource)
+          .build();
+
+      // Wait for the create operation to complete.
+      Operation response = instancesClient.insertAsync(insertInstanceRequest).get();
+
+      if (response.hasError()) {
+        System.out.println("Instance creation failed ! ! " + response);
```

would it be better to throw an exception here?

---

That would be nice. But we thought of a graceful exception handling and hence followed this style across all previous samples.